app/(app)/offence/index.tsx

```diff
@@ -41,52 +46,74 @@ const OffencePage = () => {
-              hasError={isTouched && !offenceType}
-              value={offenceType ? getOffenceTypeLabel(offenceType) : undefined}
-              placeholder="Vyberte druh priestupku"
+      <ScrollView alwaysBounceHorizontal={false}>
+        <ScreenContent>
+          <Field label="Vozidlo">
+            <TextInput
+              value={ecv}
+              isDisabled={!!role?.actions.scanCheck}
+              onChangeText={(value) => setState({ ecv: value })}
             />
-          </Link>
-        </Field>
-
-        <Field
-          label="Vyriešenie priestupku"
-          errorMessage={isTouched && !resolutionType ? requiredText : undefined}
-        >
-          <Link asChild href="/offence/resolution-type">
-            <SelectButton
-              hasError={isTouched && !resolutionType}
-              value={resolutionType ? getResolutionTypeLabel(resolutionType) : undefined}
-              placeholder="Vyberte druh vyriešenia"
-            />
-          </Link>
-        </Field>
-
-        <SelectRow
-          disabled={!role?.actions.subjective}
-          label="Objektívna zodpovednosť"
-          onValueChange={() => setState({ isObjectiveResponsibility: !isObjectiveResponsibility })}
-          value={!!isObjectiveResponsibility}
-        />
-
-        <ContinueButton loading={isSubmitting} disabled={isSubmitting} onPress={onSubmit} />
-      </ScreenContent>
+          </Field>
+
+          <Field label="Poloha">
+            <PressableStyled
+              onPress={() => {
+                router.push('/offence/location')
+              }}
+            >
+              <LocationMapPreview />
+            </PressableStyled>
+          </Field>
+
+          <Field
+            label="Druh priestupku"
+            errorMessage={isTouched && !offenceType ? requiredText : undefined}
+          >
+            <Link asChild href="/offence/offence-type">
+              <SelectButton
+                hasError={isTouched && !offenceType}
+                value={offenceType ? getOffenceTypeLabel(offenceType) : undefined}
+                placeholder="Vyberte druh priestupku"
+              />
+            </Link>
+          </Field>
+
+          <Field
+            label="Vyriešenie priestupku"
+            errorMessage={isTouched && !resolutionType ? requiredText : undefined}
+          >
+            <Link asChild href="/offence/resolution-type">
+              <SelectButton
+                hasError={isTouched && !resolutionType}
+                value={resolutionType ? getResolutionTypeLabel(resolutionType) : undefined}
+                placeholder="Vyberte druh vyriešenia"
+              />
+            </Link>
+          </Field>
+
+          <SelectRow
+            disabled={!role?.actions.subjective}
+            label="Objektívna zodpovednosť"
+            onValueChange={() =>
+              setState({ isObjectiveResponsibility: !isObjectiveResponsibility })
+            }
+            value={!!isObjectiveResponsibility}
```

Comment:
I would think about if we are not able to set `isObjectiveResponsibility` to `false` in the provider, so it'll never be `undefined.  Obviously, it's okay to display `undefined` and `false` the same way in UI, so it's probably okay to set it as default in provider.